### PR TITLE
This is the initial work to get payments working with Stripe

### PR DESCRIPTION
**Pull request by tbrown1979**

### CODE REVIEW

File: src/main/scala/com/speedrunga/Server.scala [hunk 1]
```diff
@@ -35,34 +39,35 @@ object Server {
...
+      _                       <- Stream.eval(Migrator.migrate[F](config.postgres))
+      transactor              <- buildTransactor[F](config.postgres)
+      postgresRepo             = PostgresDoobieRepository.build[F](transactor)
+      swagSupport              = SwaggerSupport.apply[F]
+      middleware               = swagSupport.createRhoMiddleware()
+      helloSvc                 = new MyRoutes(swagSupport).toRoutes(middleware)
+      routes                   = GameRoutes.routes(postgresRepo) <+> BountyRoutes.routes(postgresRepo) <+> CategoryRoutes.routes(postgresRepo)
+      config                  <- Stream.eval(loadConfig[F])
+      stripeClient            <- Stream.resource(StripeClient.build[F](APIKey(config.apiKey), postgresRepo, config.assetsHost, config.host, postgresRepo))
+//      webhookMiddleware        = StripeSignatureVerifier.middleware[F](SigningSecret(config.signingSecret), Tolerance(5 minutes))
```
Comment (browncap) [inline src/main/scala/com/speedrunga/Server.scala]: need?

File: src/main/scala/com/speedrunga/Server.scala [hunk 2]
```diff
@@ -35,34 +39,35 @@ object Server {
...
-      middleware         = swagSupport.createRhoMiddleware()
-      helloSvc           = new MyRoutes(swagSupport).toRoutes(middleware)
-      routes             = GameRoutes.routes(postgresRepo) <+> BountyRoutes.routes(postgresRepo) <+> CategoryRoutes.routes(postgresRepo)
-      config            <- Stream.eval(loadConfig[F])
-      stripeClient      <- Stream.resource(StripeClient.build[F](APIKey(config.apiKey), postgresRepo))
-      webhookMiddleware  = StripeSignatureVerifier.middleware[F](SigningSecret(config.signingSecret), Tolerance(5 minutes))
-      paymentRoutes      = PaymentRoutes.routes[F](stripeClient)
-      exitCode          <-
+      config                  <- Stream.eval(loadConfig[F])
+      implicit0(logger:       SelfAwareStructuredLogger[F]) <- Stream.eval(Slf4jLogger.create[F])
```
Comment (browncap) [inline src/main/scala/com/speedrunga/Server.scala]: this formatting is intentional?

File: src/main/scala/com/speedrunga/stripe/PaymentService.scala [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                                    F.fromOption(maybeCategory, new RuntimeException("No category found"))
+                                  }
+          _                  <- L.debug(s"Pending bounty: ${paymentSucceeded.data.`object`.id}")
+          nowMs              <- C.realTime(TimeUnit.MILLISECONDS)
+          bounty             =  Bounty(
+                                  game.gameId,
+                                  BountyId(pendingBounty.pendingBountyId.show),
+                                  pendingBounty.categoryId,
+                                  pendingBounty.creator,
+                                  Email("Gotta get that email"),
```
Comment (browncap) [inline src/main/scala/com/speedrunga/stripe/PaymentService.scala]: now or later?

File: src/main/scala/com/speedrunga/stripe/StripeClient.scala [hunk 4]
```diff
@@ -19,24 +24,32 @@ import org.http4s.implicits._
 import scala.concurrent.ExecutionContext
 
 trait StripeClient[F[_]] {
-  def createPaymentIntent(amount: Money, currency: String): F[PaymentIntent]
-  def createSession(gameId: GameId, amount: Money, currency: String): F[CheckoutSessionId]
+//  def createPaymentIntent(amount: Money, currency: String): F[PaymentIntent]
```
Comment (browncap) [inline src/main/scala/com/speedrunga/stripe/StripeClient.scala]: remove?

File: src/main/scala/com/speedrunga/stripe/StripeClient.scala [hunk 5]
```diff
@@ -19,24 +24,32 @@ import org.http4s.implicits._
...
-          val req =
-            Request[F](
-              method = Method.POST,
-              uri = uri"https://api.stripe.com/v1/payment_intents",
-              headers = Headers.of(Authorization(Credentials.Token(AuthScheme.Bearer, apiKey.value)))
-            ).withEntity(UrlForm("amount" -> amount.show, "currency" -> "usd", "payment_method_types[]" -> "card"))
-
-          client.expect[PaymentIntent](req)
-        }
+//        override def createPaymentIntent(amount: Money, currency: String): F[PaymentIntent] = {
```
Comment (browncap) [inline src/main/scala/com/speedrunga/stripe/StripeClient.scala]: remove?

File: src/main/scala/com/speedrunga/stripe/PaymentService.scala (hunk 3 above)
Comment (tbrown1979) [reply src/main/scala/com/speedrunga/stripe/PaymentService.scala]: I'm going to wait on this. If we end up making accounts this probably isn't even necessary on this model.

**Pull request merged**